First pull request

kgabinete08 opened the pull request:
Closely followed the solution videos so far.  Grateful for your input.

knwang committed:
Base Project

knwang committed:
Use Ruby 2.0.0

poiyzy committed:
Updated bootstrap sass to 3.0

knwang committed:
Merge pull request #4 from poiyzy/update_bootstrap_sass_to_3.0

Updated bootstrap sass to 3.0

knwang committed:
Upgrade to Rails 4

knwang committed:
Add better_errors

knwang committed:
Add rails_12factor

knwang committed:
Add Ruby version

knwang committed:
Add bin directory

poiyzy committed:
Fixed Remove Icon displaying in bootstrap 3.

knwang committed:
Merge pull request #15 from poiyzy/fix-icon-display-in-bootstarp3

Fixed Remove Icon displaying in bootstrap 3.

knwang committed:
Bump to Ruby 2.1.1

knwang committed:
Bump rspec and Rails

knwang committed:
Lock database cleaner to 1.2.0

knwang committed:
Update sass-rails

knwang committed:
Use rspec-rails 2.99

knwang committed:
Include pry in test

knwang committed:
Use Postgres database locally

knwang committed:
Update database.yml

kgabinete08 committed:
created video mvc and routes

kgabinete08 committed:
added categories

kgabinete08 committed:
added tests for video and category models

tomtomecek commented on app/models/category.rb:
```diff
@@ -0,0 +1,5 @@
+class Category < ActiveRecord::Base
+  has_many :videos, -> { order("title")}
```
You lack a space after title bracket, but this can also be written using a symbol - `-> { order(:title) }`

tomtomecek commented on app/views/categories/show.html.haml:
```diff
@@ -0,0 +1,8 @@
+    - @category.videos.each do |video|
+      .video.col-sm-2
+        = link_to video, nil do
+          = image_tag(video.small_cover_url)
```
Good job using image_tag :) A lot of times you will see it without brackets though :) pretty much like link_to helper.

tomtomecek commented on app/views/videos/index.html.haml:
```diff
@@ -0,0 +1,9 @@
+%article.video_category
+  - @categories.each do |category|
+    %header
+      %h3= link_to category.name, category
```
I like shortened named_routes syntax :) good job.

tomtomecek commented on config/database.yml:
```diff
@@ -0,0 +1,15 @@
+development:
```
Normally this file is in `.gitignore`, and the example is for you to build it according to that. This may require a username though and that is not a public detail. You can overcome it with `<%= ENV["USER"] %>`. You can use ERB in yaml files here.

tomtomecek commented on config/routes.rb:
```diff
@@ -1,3 +1,6 @@
 Myflix::Application.routes.draw do
-  get 'ui(/:action)', controller: 'ui'
```
You are not going to use UI controller anymore? :)

tomtomecek commented on db/seeds.rb:
```diff
@@ -5,3 +5,12 @@
+comedy = Category.create!(name: 'Comedy')
+drama = Category.create!(name: 'Drama')
+
+Video.create!(title: 'Family Guy', description: 'Brian makes Stewie watch viral video.', small_cover_url: '/tmp/family_guy.jpg', category: comedy)
```
Slightly better way how to write this would be a different format for your hash like this:

``` ruby
Video.create!(title: "...",
              description: "...",
              small_cover_url: "...",
              another_url: "...")
```

tomtomecek commented on db/seeds.rb:
```diff
@@ -5,3 +5,12 @@
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+
+comedy = Category.create!(name: 'Comedy')
```
Good job using `.create!`

tomtomecek commented on spec/models/category_spec.rb:
```diff
@@ -0,0 +1,5 @@
+require 'spec_helper'
+
+describe Category do
+  it { should have_many(:videos)}
```
You lack a space after `:videos) }`

tomtomecek commented on spec/models/video_spec.rb:
```diff
@@ -0,0 +1,7 @@
+require 'spec_helper'
+
+describe Video do
+  it { should belong_to(:category)}
```
Lacking a closing space in hash. Also for lines after this.

kgabinete08 committed:
cosmetic changes, ui route added

kgabinete08 commented:
- Added the missing spaces before closing bracket
- added ui route
- added database.yml to .gitignore, also removed database.yml from remote repo.  This may have caused some conflict and will need to attempt to merge manually.

tomtomecek commented:
You have to remove the database.yml your local git repo and push that to update remote repo. I guess that would be the easiest change. Certainly do not delete files in remote repo and try to pull it back.

kgabinete08 commented:
That makes sense, I'll do that.  So in my commit, after database.yml has been recreated then added in .gitignore, it won't be added in the commit anymore.  I should be safe to merge after that?

tomtomecek commented:
yea, its not a big deal at this moment as you have no confidential data like user or password there, but it is that kind of file that should be in a .gitignore :)

kgabinete08 commented:
Thanks Tomas, I'll make sure to keep that it mind.  Pull request is now merged. :)
